@here/harp-vectortile-datasource/lib/adapters/omv/OmvDataAdapter.ts

```diff
@@ -460,27 +512,58 @@ export class OmvDataAdapter implements DataAdapter, OmvVisitor {
 
         checkWinding(geometry);
 
-        const env = createFeatureEnv(
-            this.m_layer,
-            feature,
-            "polygon",
-            storageLevel,
-            this.m_processor.storageLevelOffset,
-            this.m_logger
-        );
-
         this.m_processor.processPolygonFeature(
             layerName,
             layerExtents,
             geometry,
-            env,
-            storageLevel
+            properties,
+            decodeFeatureId(feature, properties, logger)
         );
     }
 
+    /**
+     * Applies any filter and modifiers to a given feature.
+     *
+     * @param feature - The feature to filter and modify.
+     * @param filterFunc - The filtering function.
+     * @param modifierFunc - The modifier function.
+     * @returns The modified feature properties or `undefined` if feature is filtered out.
+     */
+    private filterAndModifyFeature(
+        feature: com.mapbox.pb.Tile.IFeature,
+        filterFunc?: (...args: any[]) => boolean,
+        modifierFunc?: (...args: any[]) => boolean
+    ): ValueMap | undefined {
+        const storageLevel = this.m_tileKey.level;
+        const layerName = this.m_layer.name;
+        const geometryType = asGeometryType(feature);
+
+        if (
+            this.m_dataFilter &&
+            filterFunc!.apply(this.m_dataFilter, [layerName, geometryType, storageLevel]) === false
+        ) {
+            return undefined;
+        }
+
+        const properties = readAttributes(this.m_layer, feature);
+        const env = new MapEnv(properties);
+        if (
+            this.m_featureModifiers?.find(fm => {
+                // TODO: The logic of feature ignore should be actually in the feature filtering
+                // mechanism - see OmvFeatureFilter.
+                assert(modifierFunc !== undefined);
+                return !modifierFunc!.apply(fm, [layerName, env, this.m_tileKey.level]);
```

Comment:
this is some genius level typescript, I tried for an hour to figure it out and I just understand...

Reply:
Well, I found it is possible to do it by setting:

modifierName: "doProcessPointFeature" | "doProcessLineFeature" | "doProcessPolygonFeature",

And then go `fm[modifierName]`

Reply:
However.... that isn't so nice to have to hard code the methods there. I tried to find a way to extract it from the OmvFeatureModifier but to no avail

Reply:
Nevermind,

Reply:
ok